paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -1145,8 +1148,28 @@ public final void createPartition(
             CatalogPartitionSpec partitionSpec,
             CatalogPartition partition,
             boolean ignoreIfExists)
-            throws CatalogException {
-        throw new UnsupportedOperationException();
+            throws CatalogException, PartitionAlreadyExistsException {
+        if (partitionExists(tablePath, partitionSpec)) {
+            if (!ignoreIfExists) {
+                throw new PartitionAlreadyExistsException(getName(), tablePath, partitionSpec);
+            }
+        }
+        try {
+            if (catalog instanceof CachingCatalog) {
+                catalog = ((CachingCatalog) catalog).wrapped();
+            }
+
+            if (catalog instanceof HiveCatalog) {
+                Identifier identifier = toIdentifier(tablePath);
+                catalog.createPartition(identifier, partitionSpec.getPartitionSpec());
```

Comment:
Just invoke this?

Reply:
Had move the logic to AbstractCatalog, so Catalog API not contains createPartition API.